config/configauth/authenticator.go

```diff
@@ -92,3 +74,21 @@ func defaultStreamInterceptor(srv interface{}, stream grpc.ServerStream, info *g
 
 	return handler(srv, stream)
 }
+
+var authRegistry = map[string]Authenticator{}
+
+func AddAuthenticatorToRegistry(name string, auth Authenticator) error {
+	if _, ok := authRegistry[name]; ok {
+		return errors.New(fmt.Sprintf("Authenticator with name %s already registered", name))
+	}
+	authRegistry[name] = auth
+	return nil
+}
+
+func GetAuthenticatorFromRegistry(name string) (Authenticator, error) {
```

Comment:
should this be `(Authenticator, bool)` instead? The error could be created in configrpc.go

Reply:
Yeah it can be, I wasn't sure if there was a case where we would catch the `false` value and not worry about it instead of just bubbling up the error. At least this way I don't have to create a new error elsewhere